nwbinspector/checks/tables.py

```diff
@@ -194,7 +194,7 @@ def check_table_values_for_dict(table: DynamicTable, nelems: int = 200):
 def check_col_not_nan(table: DynamicTable, nelems: Optional[int] = 200):
     """Check if all of the values in a single column of a table are NaN."""
     for column in table.columns:
-        if not hasattr(column, "data") or isinstance(column, VectorIndex) or isinstance(column.data[0], str):
+        if not hasattr(column, "data") or isinstance(column, VectorIndex) or not isinstance(column[0], Real):
```

Comment:
```suggestion
        if not hasattr(column, "data") or isinstance(column, VectorIndex) or column.dtype == np.dtype('float'):
```

Reply:
A couple things here - the original `isinstance(column[0], Real)` was allowing integer types for which `NaN` has no meaning (and if a list with np.nan is casted as a numpy array with integer dtype, it collapses to `0`), so removed that.

For the suggested `column.dtype` approach,

1. the `.dtype` attribute may not be set to the `VectorData` `data` object if the NWBFile is still in-memory (only if it is being read from disk). Minor workaround to that is just always cast the first element as an array to get the `dtype`

2. `that_dtype == np.dtype("float")` is a strict assertion to exactly `float64`, whereas any other `floatXYZ` type would have valid interpretation of `np.nan`

This is also an early skip condition, so logic for detection is negated.

`not np.issubdtype(np.array(column[0]).dtype, np.floating)` is the current approach I'm taking, which is valid for all numpy-supported float types (including `float128`).